glide-core/src/client/mod.rs

```diff
@@ -246,6 +246,50 @@ impl Client {
         .boxed()
     }
 
+    // Cluster scan is not passed to redis-rs as a regular command, so we need to handle it separately.
+    // We send the command to a specific function in the redis-rs cluster client, which internally handles the
+    // the complication of a command scan, and generate the command base on the logic in the redis-rs library.
+    //
+    // The function returns a tuple with the cursor and the keys found in the scan.
+    // The cursor is not a regular cursor, but an ARC to a struct that contains the cursor and the data needed
+    // to continue the scan called ScanState.
+    // In order to avoid passing Rust GC to clean the ScanState when the cursor (ref) is passed to the wrapper,
+    // which means that Rust layer is not aware of the cursor anymore, we need to keep the ScanState alive.
+    // We do that by storing the ScanState in a global container, and return a cursor-id of the cursor to the wrapper.
+    //
+    // The wrapper create an object contain the cursor-id with a drop function that will remove the cursor from the container.
+    // When the ref is removed from the hash-map, there's no more references to the ScanState, and the GC will clean it.
+    pub async fn cluster_scan<'a>(
+        &'a mut self,
+        scan_state_cursor: &'a ScanStateRC,
+        match_pattern: &'a Option<&str>,
+        count: Option<usize>,
+        object_type: Option<ObjectType>,
+    ) -> RedisResult<Value> {
+        match self.internal_client {
+            ClientWrapper::Standalone(_) => {
+                unreachable!("Cluster scan is not supported in standalone mode")
+            }
+            ClientWrapper::Cluster { ref mut client } => {
+                let (cursor, keys) = client
+                    .cluster_scan(
+                        scan_state_cursor.clone(),
+                        *match_pattern,
+                        count,
+                        object_type,
+                    )
+                    .await?;
+
+                let cluster_cursor_id = if cursor.is_finished() {
+                    Value::BulkString("finished".into())
```

Comment:
This should be made a constant so that it's accessible/consistent for all languages

Reply:
Please explain.

Reply:
This "finished" string a special string that the Rust size uses to tell the caller that there's no more data to request. We should propagate this constant out to the API layer so that all languages look for the same thing, rather than having each language hard-code this constant.

If the constant changes for whatever reason then all languages will break if this isn't done.

Reply:
Gotcha 
Agree, will do